source/Octopus.Tentacle.Tests.Integration/Support/TentacleTestCaseSourceAttribute.cs

```diff
@@ -0,0 +1,378 @@
+
+                if (source != null)
+                {
+                    foreach (object? item in source)
+                    {
+                        // First handle two easy cases:
+                        // 1. Source is null. This is really an error but if we
+                        //    throw an exception we simply get an invalid fixture
+                        //    without good info as to what caused it. Passing a
+                        //    single null argument will cause an error to be
+                        //    reported at the test level, in most cases.
+                        // 2. User provided an ITestCaseData and we just use it.
+                        ITestCaseData? parms = item == null
+                            ? new TestCaseParameters(new object?[] { null })
+                            : item as ITestCaseData;
+
+                        if (parms == null)
+                        {
+                            object?[]? args = null;
+
+                            // 3. An array was passed, it may be an object[]
+                            //    or possibly some other kind of array, which
+                            //    TestCaseSource can accept.
+                            var array = item as Array;
+                            if (array != null)
+                            {
+                                // If array has the same number of elements as parameters
+                                // and it does not fit exactly into single existing parameter
+                                // we believe that this array contains arguments, not is a bare
+                                // argument itself.
+                                var parameters = method.GetParameters();
+                                var argsNeeded = parameters.Length;
+                                if (argsNeeded > 0 && argsNeeded == array.Length && parameters[0].ParameterType != array.GetType())
+                                {
+                                    args = new object?[array.Length];
+                                    for (var i = 0; i < array.Length; i++)
+                                        args[i] = array.GetValue(i);
+                                }
+                            }
+
+                            if (args == null)
+                            {
+                                args = new object?[] { item };
+                            }
+
+                            parms = new TestCaseParameters(args);
+                        }
+
+                        // If the test is trying to test .net6.0 TentacleClient against .net48 Tentacle,
+                        // then add the 'Net60ClientNet48Tentacle' category so we can potentially run these tests in a different test run
+                        #if !NETFRAMEWORK
+                        if (item is TentacleConfigurationTestCase {TentacleRuntime: TentacleRuntime.Framework48} testCase)
```

Comment:
This is the logic for adding the custom category. It only gets added if:

1. Running in .net 6.0
2. The net48 Tentacle is explicitly defined in the test case